apollo-client/src/main/java/com/ctrip/framework/apollo/internals/AbstractConfig.java

```diff
@@ -453,14 +462,28 @@ public void run() {
 
   private boolean isConfigChangeListenerInterested(ConfigChangeListener configChangeListener, ConfigChangeEvent configChangeEvent) {
     Set<String> interestedKeys = m_interestedKeys.get(configChangeListener);
+    Set<String> interestedKeyPatterns = m_interestedKeyPatterns.get(configChangeListener);
 
-    if (interestedKeys == null || interestedKeys.isEmpty()) {
+    if ((interestedKeys == null || interestedKeys.isEmpty())
+        && (interestedKeyPatterns == null || interestedKeyPatterns.isEmpty())) {
       return true; // no interested keys means interested in all keys
     }
 
-    for (String interestedKey : interestedKeys) {
-      if (configChangeEvent.isChanged(interestedKey)) {
-        return true;
+    if (interestedKeys != null) {
+      for (String interestedKey : interestedKeys) {
+        if (configChangeEvent.isChanged(interestedKey)) {
+          return true;
+        }
+      }
+    }
+
+    if (interestedKeyPatterns != null) {
+      for (String interestedKeyPattern : interestedKeyPatterns) {
+        for (final String changedKey : configChangeEvent.changedKeys()) {
+          if (changedKey.matches(interestedKeyPattern)) {
```

Comment:
looks like the regex match is kind of slow? Is it worth to provide such powerful capability for the sake of performance? Or simply provide a `prefix match` capability?

Reply:
Reasonable

Reply:
When we use Ribbon special client config setting,key-prefix doesn't support perfectly.

Reply:
> When we use Ribbon special client config setting,key-prefix doesn't support perfectly.

Would you please provide some example?

Reply:
If we have two ribbon client named order and goods,the setting would be like this:
`order.ribbon.ReadTimeout=3000`
`order.ribbon.MaxAutoRetriesNextServer=0`
`goods.ribbon.ReadTimeout=1000`
`goods.ribbon.MaxAutoRetriesNextServer=1`
If i want to listening any change about order or goods,i must create two listeners.But if key-pattern works(^.+.ribbon..+$),i just only need one listener.

Reply:
Thanks for providing the example, I believe some users would have the same scenarios. However, I would suggest listening all the keys and use regex to match the ribbon pattern in your business logic for now, which is clear and easy.